### DISCUSSION TITLE
Fancy commit labels

### DISCUSSION THREAD

**Discussion by iboB** [community/community#139568] [Repositories] [answered]
### Select Topic Area

Question

### Body

I just saw that one of my commits has received a fancy label in GitHub:
![image](https://github.com/user-attachments/assets/f520a793-464f-4821-8d7b-d2dd53803041)

The actual text is `test: restore itlib c++20 test`, so `test:` turned into this label box. The div of the box is called `IssueLabel`, but my repo does not have an issue label `Test`.

I didn't know we could do that with commits. 

How does this work? Is it documented anywhere?

**Comment by ghasemi67** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> I just saw that one of my commits has received a fancy label in GitHub:
> ![image](https://github.com/user-attachments/assets/f520a793-464f-4821-8d7b-d2dd53803041)
> 
> The actual text is `test: restore itlib c++20 test`, so `test:` turned into this label box. The div of the box is called `IssueLabel`, but my repo does not have an issue label `Test`.
> 
> I didn't know we could do that with commits. 
> 
> How does this work? Is it documented anywhere?
>

**Comment by lovrozagar** [comment, marked as answer]
Hi iboB,

What you're seeing is a feature of GitHub that automatically detects conventional commit messages. When you use a format like test: in your commit message, GitHub interprets it as a scope and generates a label based on that prefix.

This behavior is often linked to projects that follow the Conventional Commits specification, which helps in organizing commits and generating changelogs. The labels you see in the UI are dynamically generated from the commit prefixes and are not tied to issue labels in your repository.

While GitHub doesn't have specific documentation on this feature, you can learn more about Conventional Commits and how it's used in release management here.

Let me know if you have any other questions!
